includes/GbWeiss.php

```diff
@@ -231,7 +239,39 @@ public function wooCommerceOrderStatusChanged(int $orderId, string $from, string
             return;
         }
 
-        // Send request.
+        $this->updateAuthToken();
+        $this->createLogisticsOrderAndUpdateOrderState($order);
+    }
+
+    /**
+     * Creates a logistics order using the Gebrueder Weiss API and updates the status of the WooCommerce order.
+     *
+     * @param object $order The WooCommerce order.
+     * @return void
+     */
+    public function createLogisticsOrderAndUpdateOrderState(object $order)
+    {
+        $logisticsOrder = LogisticsOrderFactory::buildFromWooCommerceOrder($order);
+        $authToken = $this->settingsRepository->getAccessToken();
+        $this->writeApiClient->getConfig()->setAccessToken($authToken);
+
+        try {
+            $this->writeApiClient->logisticsOrderPost($logisticsOrder);
+            $order->set_status("on-hold");
+            $order->save();
+        } catch (ApiException $exception) {
+            if ($exception->getCode() === 400) {
```

Comment:
you could extract the code only one time

Reply:
For me it is easier to read this if than to read something like `if(excpetionHasStatusCode($exception, 400) { .. }`.